deploy.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -e
+
+VERSION=`date "+%Y%m%dt%H%M%S"`
+LOGFILE="$VERSION.log"
+
+MIGRATE="--migrate"
+if [ "$GCLOUD_PROJECT" = "dartlang-pub-dev" ]; then
+  MIGRATE=""
+fi
+
+ANALYZER_HOST="https://""$VERSION""-dot-analyzer-dot-""$GCLOUD_PROJECT"".appspot.com"
+SEARCH_HOST="https://""$VERSION""-dot-search-dot-""$GCLOUD_PROJECT"".appspot.com"
+APP_HOST="https://""$VERSION""-dot-""$GCLOUD_PROJECT"".appspot.com"
+
+if [ "$1" = "all" ] || [ "$1" = "analyzer" ]; then
+  OLD_ANALYZER=`gcloud app versions list --service analyzer --format='value(id)'`
+
+  echo "Deploying analyzer ($VERSION)..."
+  echo "Deploying analyzer ($VERSION)..." >>$LOGFILE
+  gcloud app deploy analyzer.yaml --no-promote -v $VERSION -q >>$LOGFILE 2>&1
+
+  echo "Checking analyzer status on $ANALYZER_HOST ..."
+  echo "Checking analyzer status on $ANALYZER_HOST ..." >>$LOGFILE
+  curl "$ANALYZER_HOST""/debug" -s -f  >/dev/null
+
+  echo "Migrating traffic on analyzer..."
+  echo "Migrating traffic on analyzer..." >>$LOGFILE
+  gcloud app services set-traffic analyzer --splits $VERSION=1 $MIGRATE -q >>$LOGFILE 2>&1
```

Comment:
Is there a reason why you want to split traffic?

So far I've always manually checked that the deployed version works and then migrated all of the traffic to the new version.

If one splits the traffic than it might take longer for us to see if a version has bugs, since only a subset of the IPs are routed to the newly deployed version. Also reproducing user reported bugs is more difficult due to IP based traffic splitting.